R/polars_envvars.R

```diff
@@ -0,0 +1,134 @@
+#' * `POLARS_FMT_TABLE_ROUNDED_CORNERS` (`"0"`): Apply rounded corners to
+#'   UTF8-styled tables (only applies to UTF8 formats).
+#' * `POLARS_MAX_THREADS` (`<variable>`): Maximum number of threads used to
+#'   initialize the thread pool. The thread pool is locked once polars is loaded,
+#'   so this envvar must be set before loading the package.
+#' * `POLARS_STREAMING_CHUNK_SIZE` (`<variable>`): Chunk size used in the
+#'   streaming engine. Integer larger than 1. By default, the chunk size is
+#'   determined by the schema and size of the thread pool. For some datasets
+#'   (esp. when you have large string elements) this can be too optimistic and
+#'   lead to Out of Memory errors.
+#' * `POLARS_TABLE_WIDTH` (`<variable>`): Set the maximum width of a table in
+#'   characters.
+#' * `POLARS_VERBOSE` (`"0"`): Enable additional verbose/debug logging.
+#' * `POLARS_WARN_UNSTABLE` (`"0"`): Issue a warning when unstable functionality
+#'    is used. Enabling this setting may help avoid functionality that is still
+#'    evolving, potentially reducing maintenance burden from API changes and bugs.
+#'    Can be `"0"` or `"1"`.
+#'
+# See https://github.com/pola-rs/polars/blob/6913b16c255f38c8609ea11f1b2edab6482cedc5/py-polars/polars/config.py#L73
+#' The following configuration options are present in the Python API but
+#' currently cannot be changed in R: decimal separator, thousands separator,
+#' float precision, float formatting, trimming decimal zeros.
+#'
+#' @return `polars_envvars()` returns a named list where the names are the names
+#'   of environment variables and values are their values.
+#'
+#' @export
+#' @examples
+#' polars_envvars()
+#'
+#' pl$DataFrame(x = "This is a very very very long sentence.")
+#'
+#' Sys.setenv(POLARS_FMT_STR_LEN = 50)
+#' pl$DataFrame(x = "This is a very very very long sentence.")
+#'
+#' # back to default
+#' Sys.setenv(POLARS_FMT_STR_LEN = 32)
+#'
+polars_envvars = function() {
+  envvars = rbind(
+    # c("POLARS_AUTO_STRUCTIFY", ""),
+    c("POLARS_FMT_MAX_COLS", "5"),
+    c("POLARS_FMT_MAX_ROWS", "8"),
+    # Exist in polars but can't be set (even in py-polars)
+    # c("POLARS_FMT_NUM_DECIMAL", ""),
+    # c("POLARS_FMT_NUM_GROUP_SEPARATOR", ""),
+    # c("POLARS_FMT_NUM_LEN", ""),
+    c("POLARS_FMT_STR_LEN", "32"),
+    c("POLARS_FMT_TABLE_CELL_ALIGNMENT", "LEFT"),
+    c("POLARS_FMT_TABLE_CELL_LIST_LEN", "3"),
+    c("POLARS_FMT_TABLE_CELL_NUMERIC_ALIGNMENT", "LEFT"),
+    c("POLARS_FMT_TABLE_DATAFRAME_SHAPE_BELOW", "0"),
+    c("POLARS_FMT_TABLE_FORMATTING", "UTF8_FULL_CONDENSED"),
+    c("POLARS_FMT_TABLE_HIDE_COLUMN_DATA_TYPES", "0"),
+    c("POLARS_FMT_TABLE_HIDE_COLUMN_NAMES", "0"),
+    c("POLARS_FMT_TABLE_HIDE_COLUMN_SEPARATOR", "0"),
+    c("POLARS_FMT_TABLE_HIDE_DATAFRAME_SHAPE_INFORMATION", "0"),
+    c("POLARS_FMT_TABLE_INLINE_COLUMN_DATA_TYPE", "0"),
+    c("POLARS_FMT_TABLE_ROUNDED_CORNERS", "0"),
+    c("POLARS_MAX_THREADS", unwrap(get_global_rpool_cap())$capacity),
```

Comment:
@etiennebacher Isn't this `threadpool_size`?

Reply:
Ah right I'll fix this